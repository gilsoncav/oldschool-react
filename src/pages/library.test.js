import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Library from './library';

const mockStore = configureStore();

describe('Library page component', () => {
  it('renders correctly when loading', () => {
    const store = mockStore({
      games: {
        isLoadingGameList: true,
        gameList: [],
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <Library />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when not loading and has games', () => {
    const store = mockStore({
      games: {
        isLoadingGameList: false,
        gameList: [
          {
            VideoURL: undefined,
            Name: 'Game',
            Platform: 'Game',
            Overview: 'Overview',
          },
        ],
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <Library />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when not loading and not has games', () => {
    const store = mockStore({
      games: {
        isLoadingGameList: false,
        gameList: [],
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <Library />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
