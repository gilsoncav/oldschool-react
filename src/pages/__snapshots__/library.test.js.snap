// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library page component renders correctly when loading 1`] = `
Array [
  .c0 h1 {
  margin-top: 25px;
  text-align: center;
  color: #ff6347;
}

.c0 .ant-col {
  margin-top: 20px;
  margin-bottom: 20px;
  height: 470px;
  min-height: 470px;
  max-height: 470px;
}

<div
    style={
      Object {
        "height": "100vh",
        "overflow": "auto",
      }
    }
  >
    <div
      className="ant-row"
      style={
        Object {
          "marginTop": "2vh",
        }
      }
    >
      <div
        className="ant-col ant-col-xs-22 ant-col-xs-offset-1 ant-col-sm-16 ant-col-sm-offset-4 ant-col-md-14 ant-col-md-offset-5 ant-col-lg-12 ant-col-lg-offset-6"
      >
        <span
          className="ant-input-search game-search ant-input-search-enter-button ant-input-search-large ant-input-group-wrapper ant-input-group-wrapper-lg"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <span
            className="ant-input-wrapper ant-input-group"
          >
            <input
              className="ant-input ant-input-lg"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search Games"
              size="large"
              style={null}
              type="text"
              value=""
            />
            <span
              className="ant-input-group-addon"
            >
              <button
                className="ant-btn ant-input-search-button ant-btn-primary ant-btn-lg"
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-label="icon: search"
                  className="anticon anticon-search"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </i>
              </button>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      className="ant-row"
      style={
        Object {
          "marginLeft": -12,
          "marginRight": -12,
          "padding": "0 15px",
          "paddingBottom": "100px",
        }
      }
    >
      <div
        className="c0"
      >
        <h1>
          <div
            className="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text"
          >
            <span
              className="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
              <i
                className="ant-spin-dot-item"
              />
            </span>
            <div
              className="ant-spin-text"
            >
              Loading games...
            </div>
          </div>
        </h1>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Library page component renders correctly when not loading and has games 1`] = `
Array [
  .c1 {
  outline: none;
  background-color: #373e49;
  color: #ff6347;
  border-radius: 15px;
  height: inherit;
  padding: 10px;
  margin: 10px;
  text-align: center;
}

.c1:hover {
  background-color: #2c313a;
  cursor: pointer;
}

.c1 div:nth-child(1) div:nth-child(1) {
  width: inherit !important;
  height: 250px !important;
}

.c1 div:nth-child(2) > span,
.c1 div:nth-child(3) > span {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
}

.c1 > div:nth-child(4) > span {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.c0 h1 {
  margin-top: 25px;
  text-align: center;
  color: #ff6347;
}

.c0 .ant-col {
  margin-top: 20px;
  margin-bottom: 20px;
  height: 470px;
  min-height: 470px;
  max-height: 470px;
}

<div
    style={
      Object {
        "height": "100vh",
        "overflow": "auto",
      }
    }
  >
    <div
      className="ant-row"
      style={
        Object {
          "marginTop": "2vh",
        }
      }
    >
      <div
        className="ant-col ant-col-xs-22 ant-col-xs-offset-1 ant-col-sm-16 ant-col-sm-offset-4 ant-col-md-14 ant-col-md-offset-5 ant-col-lg-12 ant-col-lg-offset-6"
      >
        <span
          className="ant-input-search game-search ant-input-search-enter-button ant-input-search-large ant-input-group-wrapper ant-input-group-wrapper-lg"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <span
            className="ant-input-wrapper ant-input-group"
          >
            <input
              className="ant-input ant-input-lg"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search Games"
              size="large"
              style={null}
              type="text"
              value=""
            />
            <span
              className="ant-input-group-addon"
            >
              <button
                className="ant-btn ant-input-search-button ant-btn-primary ant-btn-lg"
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-label="icon: search"
                  className="anticon anticon-search"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </i>
              </button>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      className="ant-row"
      style={
        Object {
          "marginLeft": -12,
          "marginRight": -12,
          "padding": "0 15px",
          "paddingBottom": "100px",
        }
      }
    >
      <div
        className="c0"
      >
        <h1>
          Showing a maximum of 
          12
           games...
        </h1>
        <div
          className="ant-col ant-col-xs-22 ant-col-xs-offset-1 ant-col-sm-22 ant-col-sm-offset-1 ant-col-md-10 ant-col-md-offset-1 ant-col-lg-6 ant-col-lg-offset-1"
          style={
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="c1"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div>
              <div>
                <strong>
                  No video available for this game.
                </strong>
              </div>
            </div>
            <div>
              <strong>
                Title: 
              </strong>
              <span>
                Super Mario Kart
              </span>
            </div>
            <div>
              <strong>
                Platform: 
              </strong>
              <span>
                Super Nintendo Entertainment System
              </span>
            </div>
            <div>
              <strong>
                Overview: 
              </strong>
              <span>
                The best kart game in the world, y' now
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "",
]
`;

exports[`Library page component renders correctly when not loading and not has games 1`] = `
Array [
  .c0 h1 {
  margin-top: 25px;
  text-align: center;
  color: #ff6347;
}

.c0 .ant-col {
  margin-top: 20px;
  margin-bottom: 20px;
  height: 470px;
  min-height: 470px;
  max-height: 470px;
}

<div
    style={
      Object {
        "height": "100vh",
        "overflow": "auto",
      }
    }
  >
    <div
      className="ant-row"
      style={
        Object {
          "marginTop": "2vh",
        }
      }
    >
      <div
        className="ant-col ant-col-xs-22 ant-col-xs-offset-1 ant-col-sm-16 ant-col-sm-offset-4 ant-col-md-14 ant-col-md-offset-5 ant-col-lg-12 ant-col-lg-offset-6"
      >
        <span
          className="ant-input-search game-search ant-input-search-enter-button ant-input-search-large ant-input-group-wrapper ant-input-group-wrapper-lg"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <span
            className="ant-input-wrapper ant-input-group"
          >
            <input
              className="ant-input ant-input-lg"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search Games"
              size="large"
              style={null}
              type="text"
              value=""
            />
            <span
              className="ant-input-group-addon"
            >
              <button
                className="ant-btn ant-input-search-button ant-btn-primary ant-btn-lg"
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-label="icon: search"
                  className="anticon anticon-search"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </i>
              </button>
            </span>
          </span>
        </span>
      </div>
    </div>
    <div
      className="ant-row"
      style={
        Object {
          "marginLeft": -12,
          "marginRight": -12,
          "padding": "0 15px",
          "paddingBottom": "100px",
        }
      }
    >
      <div
        className="c0"
      >
        <h1>
          There are no games available, please try again later...
        </h1>
      </div>
    </div>
  </div>,
  "",
]
`;
